laurent22/joplin
# Desktop: Resolves #9927: Beta editor: Fix search results not highlighted

# Summary

- Updates `CodeMirror5Emulation` such that Joplin's CodeMirror 5 search highlighting works with CodeMirror 6.
- For now, search results aren't marked on the scrollbar &mdash; this might be something we need to implement ourselves ([see this post on the CM6 forum](https://discuss.codemirror.net/t/show-location-of-matched-search-query-in-scrollbar/6463)).

Fixes #9927.

# Testing

1. Enable the CodeMirror 5 editor
2. Search for a full-word term known to be present in multiple notes (e.g. "references")
3. Change the search term to work around #9926
4. Verify that matches are shown on the scrollbar
5. Verify that matches are highlighted in both the editor and viewer

6. Enable the CodeMirror 6 editor
7. Search for "references"
8. Change the search term to work around #9926
9. Verify that matches are highlighted in both the editor and viewer

This has been tested successfully on Ubuntu 23.10.
